x-pack/plugin/esql/src/main/antlr/EsqlBaseLexer.g4

```diff
@@ -344,19 +344,19 @@ ENRICH_FIELD_WS
     : WS -> channel(HIDDEN)
     ;
 
-// | LOOKUP ON key
+// LOOKUP ON key
 mode LOOKUP_MODE;
 LOOKUP_PIPE : PIPE -> type(PIPE), popMode;
 LOOKUP_COMMA : COMMA -> type(COMMA);
 LOOKUP_DOT: DOT -> type(DOT);
-LOOKUP_ON : 'on' -> type(ON);
+LOOKUP_ON : ON -> type(ON);
 
-LOOKUP_QUOTED_IDENTIFIER
-    : QUOTED_IDENTIFIER -> type(QUOTED_IDENTIFIER)
+LOOKUP_FIELD_ID_PATTERN
+    : ID_PATTERN -> type(ID_PATTERN)
```

Comment:
Used ID_PATTERN since it's likely in the future will use the same syntax with ENRICH (WITH foo*).
The other changes are necessary due to the way lexing works since we want to avoid ambiguities (UNQUOTED IDENTIFIER and ID_PATTERN overlap).